pkg/sbom.go

```diff
@@ -23,56 +24,113 @@ const (
 	OCI_ANNOTATION_CREATORS           = "org.spdx.creator"
 )
 
+type SPDXDocument struct {
+	// Version is the version of the SPDX specification used in the document
+	Version  string         `json:"spdxVersion"`
+	Document *v2_3.Document `json:"document"`
+}
+
 // LoadSBOMFromFile opens a file given by filename, reads its contents, and loads it into an SPDX document.
 // It also calculates the file size and generates an OCI descriptor for the file.
 // It returns the loaded SPDX document, the OCI descriptor, and any error encountered.
-func LoadSBOMFromFile(filename string) (*v2_3.Document, *ocispec.Descriptor, []byte, error) {
+func LoadSBOMFromFile(filename string, strict bool) (*SPDXDocument, *ocispec.Descriptor, []byte, error) {
 	file, err := os.Open(filename)
 	if err != nil {
 		return nil, nil, nil, err
 	}
 	defer file.Close()
 
-	return LoadSBOMFromReader(file)
+	return LoadSBOMFromReader(file, strict)
 }
 
 // LoadSBOMFromReader reads an SPDX document from an io.ReadCloser, generates an OCI descriptor for the document,
 // and returns the loaded SPDX document and the OCI descriptor.
 // If an error occurs during reading the document or generating the descriptor, the error will be returned.
-func LoadSBOMFromReader(reader io.ReadCloser) (*v2_3.Document, *ocispec.Descriptor, []byte, error) {
+func LoadSBOMFromReader(reader io.ReadCloser, strict bool) (*SPDXDocument, *ocispec.Descriptor, []byte, error) {
 	defer reader.Close()
 
 	desc, sbomBytes, err := LoadArtifactFromReader(reader, MEDIATYPE_SPDX)
 	if err != nil {
 		return nil, nil, nil, err
 	}
 
-	// Create a bytes.Reader from the slice
-	sbomReader := bytes.NewReader(sbomBytes)
-
-	// Read the SPDX document from the reader
-	doc, err := json.Read(sbomReader)
+	doc, err := getSPDXDocumentFromSBOMBytes(sbomBytes, strict)
 	if err != nil {
 		return nil, nil, nil, err
 	}
 
 	return doc, desc, sbomBytes, nil
 }
 
-// GetAnnotations returns the annotations from the SBOM
-func GetAnnotations(sbom *v2_3.Document) (map[string]string, error) {
-	var creatorstrings []string
-	for _, creator := range sbom.CreationInfo.Creators {
-		creatorstrings = append(creatorstrings, fmt.Sprint(creator.Creator))
+func getSPDXDocumentFromSBOMBytes(sbomBytes []byte, strict bool) (*SPDXDocument, error) {
+	var jsonDoc map[string]interface{}
+	err := json.Unmarshal(sbomBytes, &jsonDoc)
+	if err != nil {
+		return nil, fmt.Errorf("error unmarshaling SBOM bytes: %w", err)
+	}
+
+	version, ok := jsonDoc["spdxVersion"].(string)
+	if !ok {
+		return nil, fmt.Errorf("SBOM does not contain spdxVersion field")
+	}
+
+	sbomReader := bytes.NewReader(sbomBytes)
+	doc, err := spdxjson.Read(sbomReader)
```

Comment:
I imagine we could create new function 
`func ReadIntoWithVersion(content io.Reader, doc common.AnyDocument) (metadata Metadata, err error)` that you could then use to get the version back and insert it into your wrapper, so this should fit in nicely with what you have..

`struct Metadata { parsedVersion string }`

Thoughts @kzantow? do you know if there's a cleaner way to do this with the converter you wrote so that we can make the original Spdx version info more easily retrievable?

Reply:
Yeah I think that would work. However, in the context of this PR, I wanted to have `common.AnyDocument` have the fields that are common to all of the models such as `documentNamespace` and `SPDXID` that I could convert to a specific model depending on the version. I'm not sure how to accomplish this, but from my perspective, that would have been ideal.

Reply:
That is the intent of `spdx.Document`, to be the common document type, since all versions are backward compatible within v2, you can treat spdx.Document as a common.AnyDocument. 

> I could convert to a specific model depending on the version
is this just for output or during usage programmatically?